pyk/src/pyk/kcfg/explore.py

```diff
@@ -262,7 +262,22 @@ def extract_rule_labels(_logs: tuple[LogEntry, ...]) -> list[str]:
         # Branch
         assert len(next_states) > 1
         if all(branch_constraint for _, branch_constraint in next_states):
-            branches = [not_none(rule_predicate) for _, rule_predicate in next_states]
+            branch_preds = [flatten_label('#And', not_none(rule_predicate)) for _, rule_predicate in next_states]
+            common_preds = list(
+                unique(
+                    [
+                        pred
+                        for branch_pred in branch_preds
+                        for pred in branch_pred
+                        if all(pred in bp for bp in branch_preds)
+                    ]
+                )
+            )
+            branches = [mlAnd(pred for pred in branch_pred if pred not in common_preds) for branch_pred in branch_preds]
+            if len(common_preds) > 0:
```

Comment:
Nitpick:

```suggestion
            if common_preds:
```

(https://github.com/python/peps/blob/main/peps/pep-0008.rst?plain=1#L1470-L1485)